src/common/telemetry.ts

```diff
@@ -1,377 +1,381 @@
-                    TelemetryUtils.userType = TelemetryUtils.getUserType();
+                            case "number":
+                                measures[key] = <number>event.properties[key];
+                                break;
 
-                    Telemetry.isOptedIn = TelemetryUtils.getTelemetryOptInSetting();
-                    TelemetryUtils.saveSettings();
-                });
-            }
+                            default:
+                                properties[key] = JSON.stringify(event.properties[key]);
+                                break;
+                        }
+                    });
 
-            public static addCommonProperties(event: any): void {
-                if (Telemetry.isOptedIn) {
-                    // for the opt out event, don"t include tracking properties
-                    event.properties["RN.userId"] = TelemetryUtils.userId;
-                    event.properties["RN.machineId"] = TelemetryUtils.machineId;
+                    Telemetry.reporter.sendTelemetryEvent(event.name, properties, measures);
                 }
 
-                event.properties["RN.sessionId"] = TelemetryUtils.sessionId;
-                event.properties["RN.userType"] = TelemetryUtils.userType;
-                event.properties["RN.hostOS"] = os.platform();
-                event.properties["RN.hostOSRelease"] = os.release();
+            } catch (err) {
+                console.error(err);
             }
+        }
+    }
+
+    export function isInternal(): boolean {
+        return TelemetryUtils.userType === TelemetryUtils.USERTYPE_INTERNAL;
+    }
+
+    export function getSessionId(): string {
+        return TelemetryUtils.sessionId;
+    }
+
+    export function setSessionId(sessionId: string): void {
+        TelemetryUtils.sessionId = sessionId;
+    }
+
+    interface ITelemetrySettings {
+        [settingKey: string]: any;
+        userId?: string;
+        machineId?: string;
+        optIn?: boolean;
+        userType?: string;
+    }
+
+    class TelemetryUtils {
+        public static USERTYPE_INTERNAL: string = "Internal";
+        public static USERTYPE_EXTERNAL: string = "External";
+        public static userType: string;
+        public static sessionId: string;
+        public static optInCollectedForCurrentSession: boolean;
+
+        private static userId: string;
+        private static telemetrySettings: ITelemetrySettings = null;
+        private static TELEMETRY_SETTINGS_FILENAME: string = "VSCodeTelemetrySettings.json";
+        private static APPINSIGHTS_INSTRUMENTATIONKEY: string = "AIF-d9b70cd4-b9f9-4d70-929b-a071c400b217"; // Matches vscode telemetry key
+        private static REGISTRY_SQMCLIENT_NODE: string = "\\SOFTWARE\\Microsoft\\SQMClient";
+        private static REGISTRY_USERID_VALUE: string = "UserId";
+        private static INTERNAL_DOMAIN_SUFFIX: string = "microsoft.com";
+        private static INTERNAL_USER_ENV_VAR: string = "TACOINTERNAL";

```

Comment:
This is to let us distinguish between internal and external users. For windows machines it's done based on whether the machine is domain joined, but for macs there wasn't a standard way of determining this, so we went with defining an environment variable so we can opt-in our own machines to being an internal user and including additional info.